pkg/sfu/dependencydescriptor/bitstreamreader.go

```diff
@@ -0,0 +1,124 @@
+package dependencydescriptor
+
+import (
+	"errors"
+	"io"
+)
+
+type BitStreamReader struct {
+	buf         []byte
+	pos         int
+	remaingBits int
+}
+
+func NewBitStreamReader(buf []byte) *BitStreamReader {
+	return &BitStreamReader{buf: buf, remaingBits: len(buf) * 8}
+}
+
+func (b *BitStreamReader) RemaningBits() int {
+	return b.remaingBits
+}
+
+// Reads `bits` from the bitstream. `bits` must be in range [0, 64].
+// Returns an unsigned integer in range [0, 2^bits - 1].
+// On failure sets `BitstreamReader` into the failure state and returns 0.
+func (b *BitStreamReader) ReadBits(bits int) (uint64, error) {
+	if bits < 0 || bits > 64 {
+		return 0, errors.New("invalid number of bits, expected 0-64")
+	}
+
+	if b.remaingBits < bits {
+		b.remaingBits -= bits
+		return 0, io.EOF
```

Comment:
remaining bits goes negative here. Is that the intention? Or should it go to 0 and this should return the number of remaining bits?

Reply:
use remainingBit>=0 to indicate status is ok